azure_functions_worker/bindings/datumdef.py

```diff
@@ -92,6 +93,69 @@ def from_typed_data(cls, td: protos.TypedData):
 
         return cls(val, tt)
 
+    @classmethod
+    def from_rpc_shared_memory(
+            cls,
+            shmem: protos.RpcSharedMemory,
+            shmem_mgr) -> Optional['Datum']:
+        """
+        Reads the specified shared memory region and converts the read data into
+        a datum object of the corresponding type.
+        """
+        mem_map_name = shmem.name
+        offset = shmem.offset
+        count = shmem.count
+        data_type = shmem.type
+        ret_val = None
+        if data_type == protos.RpcDataType.bytes:
+            val = shmem_mgr.get_bytes(mem_map_name, offset, count)
+            if val is not None:
+                ret_val = cls(val, 'bytes')
+        elif data_type == protos.RpcDataType.string:
+            val = shmem_mgr.get_string(mem_map_name, offset, count)
+            if val is not None:
+                ret_val = cls(val, 'string')
+        if ret_val is not None:
+            logger.info(
+                f'Read {count} bytes from memory map {mem_map_name} '
+                f'for data type {data_type}')
+            return ret_val
+        return None
+
+    @classmethod
+    def to_rpc_shared_memory(
+            cls,
+            datum: 'Datum',
+            shmem_mgr) -> Optional[protos.RpcSharedMemory]:
+        """
+        Writes the given value to shared memory and returns the corresponding
+        RpcSharedMemory object which can be sent back to the functions host over
+        RPC.
+        """
+        if datum.type == 'bytes':
+            value = datum.value
+            shared_mem_meta = shmem_mgr.put_bytes(value)
+            data_type = protos.RpcDataType.bytes
+        elif datum.type == 'string':
+            value = datum.value
+            shared_mem_meta = shmem_mgr.put_string(value)
+            data_type = protos.RpcDataType.string
+        else:
+            raise NotImplementedError(
+                f'Unsupported datum type ({datum.type}) for shared memory'
+            )
+        if shared_mem_meta is None:
```

Comment:
I would add a few break lines (before and after this if for example) to let the code breathe a little.